Add mixed precision tests for mpp_global_sum

rem1776 opened the pull request:
**Description**
Adds i4 and i8 tests for mpp_global_sum in the test_global_minmax.F90 file and renames the file to make more sense

**How Has This Been Tested?**
make check on amd with gcc and intel

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

wrongkindofdoctor commented on test_fms/mpp/test_global_arrays.F90:
```diff
@@ -89,59 +93,87 @@ program test_global_minmax
+  allocate(dataI4_5d(isd:ied,jsd:jed, 1, 1, 1), dataI8_5d(isd:ied,jsd:jed, 1, 1, 1))
+  do i=isc, iec-1
+    do j=jsc, jec-1
+      dataI4_5d(i, j, 1, 1, 1) = dataI4(i, j) 
```
Thanks, @rem1776. This is a great start! Could you please tests that verify that dataI4_5d/I8_5d arrays with the values in a different order produce the same global sums as the original arrays per Rusty's recommendation?

mlee03 commented on test_fms/mpp/test_global_arrays.F90:
```diff
@@ -63,24 +64,27 @@ program test_global_minmax
+      dataI4(i, j) = int(rcoef * randmaxI4, kind=i4_kind)
+      dataI8(i, j) = int(rcoef * randmaxI8, kind=i8_kind)
       dataR4(i, j) = real(rcoef, kind=r4_kind)
       dataR8(i, j) = real(rcoef, kind=r8_kind)
```
Absolutely no need to change the code if you dont want to, but could we change the ordering of the loops or indices, eg to dataI4(j,i)? :D

rem1776 commented:
@wrongkindofdoctor I added the shuffleData subroutines so the data gets randomly reordered and before getting assigned to the 5D arrays and then checked with the original sums. Also changed the indices ordering for @mlee03

mlee03 commented on test_fms/mpp/test_global_arrays.F90:
```diff
@@ -0,0 +1,523 @@
+    do i=0, npes-1
+      trans = RESHAPE(shuffled(SIZE(trans)*i + 1:SIZE(trans)*(i+1)), &
+                         (/SIZE(trans,1), SIZE(trans,2) /) )
+      if(i.gt.0) then 
```
Assumes root_pe=0?

rem1776 replied:
Sorry, didn't see this till today! Can the root be any of the pe's or does it have to be the lowest value? I had thought it went from 0 to npes-1 normally. Would it be better to use `do i=root, root+npes`?

wrongkindofdoctor replied:
@rem1776 Typically, the root pe is 0, though this is not guaranteed from what I understand. `i=0-npes-1` is correct. I believe @mlee03 is referring to the check `if i .gt. 0`; it would be safer to do something like `if pe() .ne. root_pe()`

mlee03 replied:
@wrongkindofdoctor yup thank you for the clarification!

mlee03 reviewed (commented):
One minor thing :)

wrongkindofdoctor commented on test_fms/mpp/test_global_arrays.F90:
```diff
@@ -0,0 +1,523 @@
+end subroutine shuffleDataI4
+
+!> aggregates data on root and randomizes ordering, then sends partitions back to pes
+subroutine shuffleDataI8
```
The shuffleData routines do the trick, but they should accept the original arrays as arguments. It makes it easier to debug if we can tell immediately what is being passed where.

rem1776 commented:
@wrongkindofdoctor I ended up setting the root pe to 0 explicitly to avoid any others issues that came up if i changed it and also added the args in for shuffleData
